nixos/modules/system/boot/systemd-unit-options.nix

```diff
@@ -217,7 +217,7 @@ in rec {
 
     environment = mkOption {
       default = {};
-      type = types.attrs; # FIXME
+      type = with types; attrsOf (nullOr (either str package));
```

Comment:
I'm using `str | package` because `str` doesn't match derivations, which are used as values in nixpkgs, and can still be converted to strings.

Reply:
I ran an evaluation of all of the tests, nothing failed because of the more restrictive type.